### PR TITLE
Document translation of text within an image

### LINKED ISSUE #31404: [Microsoft Translator - Document Translation] API Review
New API Review meeting has been requested.

**Service Name**: Microsoft Translator - Document Translation
**Review Created By**: Hamshavathi Munibyraiah
**Review Date**: 11/13/2024 09:00 AM PT
**Release Plan**: [1500](https://web.powerapps.com/apps/821ab569-ae60-420d-8264-d7b5d5ef734c?release-plan-id=15eda0c6-ac97-ef11-8a6a-00224806dc21)
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/31384
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=ad792cde-d49a-ef11-8a6a-6045bd08e35c)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by hamshavathimunibyraiah**
Adding new option "translateTextWithImage" that can be set to true or false that enables translation of image within a document translation request.

New parameter added for batch and sync document translation.
New parameters added to TranslationStatus, DocumentStatus and DocumentTranslateContent to return the number of images translated successfully and that failed during a document translation operation.
fix https://github.com/Azure/azure-rest-api-specs/issues/31404

### CODE REVIEW

**Comment by wanlwanl** [comment]
Please set like [update tspconfig for AI.DocumentTranslation by kazrael2119 · Pull Request #31397 · Azure/azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs/pull/31397) to pass TS SDK generation.

File: specification/translation/Azure.AI.DocumentTranslation/main.tsp [hunk 1]
```diff
@@ -39,4 +39,7 @@ namespace DocumentTranslation;
 enum Versions {
   @useDependency(Azure.Core.Versions.v1_0_Preview_1)
   v2024_05_01: "2024-05-01",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  v2024_11_15_preview: "2024-11-15-preview",
```
Comment (hamshavathimunibyraiah) [inline specification/translation/Azure.AI.DocumentTranslation/main.tsp:44]: Update to the correct version: https://machinetranslation.visualstudio.com/MachineTranslation/_git/mt-api-frontend?path=/private/MTBatchService/Microsoft.MT.Batch.Frontend/Controllers/V1/ControllerConstants.cs&version=GBmaster&_a=contents

Comment (hamshavathimunibyraiah) [reply specification/translation/Azure.AI.DocumentTranslation/main.tsp]: Updated

File: specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json [hunk 2]
```diff
@@ -0,0 +1,1537 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request has succeeded.",
+            "schema": {
+              "type": "file"
+            },
+            "headers": {
+              "x-metered-usage": {
```
Comment (mikekistler) [inline specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json:628]: [⛔](https://aka.ms/azapi/guidelines#http-no-x-custom-headers) DO NOT use "x-" prefix for custom headers, unless the header already exists in production [[RFC 6648](https://datatracker.ietf.org/doc/html/rfc6648)].

File: specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json [hunk 3]
```diff
@@ -0,0 +1,1537 @@
...
+                "type": "integer",
+                "format": "int32",
+                "description": "Specifies consumption (the number of characters for which the user will be charged) for the translation job request"
+              },
+              "x-ms-client-request-id": {
+                "type": "string",
+                "format": "uuid",
+                "description": "An opaque, globally-unique, client-generated string identifier for the request."
+              },
+              "x-total-image-scans-failed": {
```
Comment (mikekistler) [inline specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json:638]: [⛔](https://aka.ms/azapi/guidelines#http-no-x-custom-headers) DO NOT use "x-" prefix for custom headers, unless the header already exists in production [[RFC 6648](https://datatracker.ietf.org/doc/html/rfc6648)].

File: specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json [hunk 4]
```diff
@@ -0,0 +1,1537 @@
...
+                "type": "integer",
+                "format": "int32",
+                "description": "Specifies the number of failed image translations within a document translation job"
+              },
+              "x-total-image-scans-succeeded": {
```
Comment (mikekistler) [inline specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json:643]: [⛔](https://aka.ms/azapi/guidelines#http-no-x-custom-headers) DO NOT use "x-" prefix for custom headers, unless the header already exists in production [[RFC 6648](https://datatracker.ietf.org/doc/html/rfc6648)].

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

This looks fine except that you should remove the "x-" from the new custom headers.

File: specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json (hunk 2 above)
Comment (hamshavathimunibyraiah) [reply specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json:638]: this header already exists in production, hence not updating.

File: specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json (hunk 3 above)
Comment (hamshavathimunibyraiah) [reply specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json]: Updated to "total-image-scans-failed"

File: specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json (hunk 4 above)
Comment (hamshavathimunibyraiah) [reply specification/translation/data-plane/Azure.AI.DocumentTranslation/preview/2024-11-01-preview/openapi.json]: Updated to "total-image-scans-succeeded"

**Comment by markcowl** [comment]
Addeed the `BreakingChangeApproved-Benign label because the change adds a new optional query parameter, there is no change in order of the query parameters, the change should not impact the order of appearance of the body parameters and the other header parameter is `clientRequestId`, which is a standard header.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**